File: tests/integration/fixtures/sample.ts
interface Greeter {
  greet(name: string): string;
}

class Hello implements Greeter {
  greet(name: string): string {
    return `Hello, ${name}!`;
  }
}

class FancyGreeter extends Hello {
  greet(name: string): string {
    return `Greetings, ${name}!`;
  }
}

function makeGreeting(name: string): string {
  const g = new Hello();
  return g.greet(name);
}

File: tests/integration/fixtures/sample.py
class Greeter:
    def greet(self, name: str) -> str:
        return f"Hello, {name}!"


class FancyGreeter(Greeter):
    def greet(self, name: str) -> str:
        return f"Greetings, {name}!"


def make_greeting(name: str) -> str:
    g = Greeter()
    return g.greet(name)
